### ISSUE TITLE
1.0rc1 tests failure with wcslib 4.24

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#3400]
Although marked as xfail, `test_radesys_defaults` and `test_radesys_defaults_full` fail with wcslib 4.24 (which is in the upcoming Debian Squeeze release):
```
____________________________ test_radesys_defaults _____________________________
   @pytest.mark.xfail(
        LooseVersion(_wcs.__version__) < LooseVersion("4.25"),
        reason="wcslib < 4.25")
    def test_radesys_defaults():
        w = _wcs.Wcsprm()
        w.ctype = ['RA---TAN', 'DEC--TAN']
        w.set()
>       assert w.radesys == "ICRS"
E       assert '' == 'ICRS'
E         + ICRS

astropy/wcs/tests/test_wcsprm.py:795: AssertionError
```
(test_radesys_defaults_full similar).

**Comment by mdboom** [comment]
I couldn't really reproduce this, but I wonder if a `wcsconfig.h` being left around from a non-system build is the problem.  Would you mind trying #3402?

**Comment by olebole** [comment]
Didn't help. However, I anyway feel a bit unlucky about the wcslib version number taken from the .h file: On Debian, the wcslib may be upgraded without upgrading astropy, and then the version number is wron. For example, in the moment, the wcslib in Debian sid is "4.24"; when I now package astropy, it will (always) return 4.24 as version number. However, once Debian squeeze is out, I will upgrade wcslib to 4.25.1 (or the version that is actual then), and then the version number in astropy._wcs and the version of the library are different (and may lead to wrong results, if anything depends on this).

I would recommend to convince Mark Calabretta to include a function returning the version number in wcslib, and to temporarily remove the wcslib version number from astropy until this happened.

**Comment by astrofrog** [comment]
Is this something we can try and fix in 1.0.1 or is it critical enough to hold back the 1.0 release?

**Comment by olebole** [comment]
The test is not critical for me; I just marked the tests as unconditionally `@pytest.mark.xfail()`. The precompiled wcs version number is, however, something I would try to avoid even in the 1.0 release, since it may give wrong results. If there is nothing internally that depends on it, I would just remove it (see my pull request).
